k8s.tf

```diff
@@ -53,3 +53,38 @@ module "eks-addons" {
   cluster_autoscaler_helm_config = var.cluster_autoscaler_helm_config
   depends_on                     = [module.eks.managed_node_groups]
 }
+
+data "aws_eks_node_group" "execnodes" {
+  cluster_name    = local.infrastructurename
+  node_group_name = regex(local.arn_to_nodegroup_name_pattern, module.eks.managed_node_groups[0]["execnodes"]["managed_nodegroup_arn"][0])[0]

```

Comment:
An alternative approach may be to set the tags for the autoscaling group `k8s.io/cluster-autoscaler/node-template/label/ `via the autoscaler helm chart: https://github.com/kubernetes/autoscaler/blob/4d8e6545032e2e3cf449c66bf142e362f12c8703/charts/cluster-autoscaler/values.yaml#L22C11-L22C11. However, what is an open question to me is how we set different labels for the different node groups. What do you think?

Reply:
> I believe the output `managed_nodegroup_id` https://github.com/aws-ia/terraform-aws-eks-blueprints/blob/8a06a6e7006e4bed5630bd49c7434d76c59e0b5e/modules/aws-eks-managed-node-groups/outputs.tf#L1 is the node group name. So my hope is to get rid of the regular expressions and the local expressions.

The id is [CLUSTER_CODE]:[NODE_GROUP_NAME], so we could use a simple replace instead of a regex.

Reply:
Yes, that would be an improvement. 
By the way: the [id](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/eks_node_group#id) is specific to the AWS provider and not the blueprints module. That is good.

Reply:
> An alternative approach may be to set the tags for the autoscaling group `k8s.io/cluster-autoscaler/node-template/label/ `via the autoscaler helm chart: https://github.com/kubernetes/autoscaler/blob/4d8e6545032e2e3cf449c66bf142e362f12c8703/charts/cluster-autoscaler/values.yaml#L22C11-L22C11. However, what is an open question to me is how we set different labels for the different node groups. What do you think?

I can't find a way to propegate the tags from the nodegroup to the asg
https://github.com/aws/containers-roadmap/issues/1541